errors.go

```diff
@@ -34,12 +34,26 @@ type Unwrappable interface {
 // Wrap annotates an error, optionally with a formatted string.
 // if %w is used the argument will be unwrapped
 func Wrap(err error, format string, args ...any) error {
-	var note string
+	if err == nil {
+		return nil
+	}
 
+	return doWrap(err, false, format, args...)
+}
+
+// QuietWrap replaces the text of the error it's wrapping.
+// if %w is used the argument will be unwrapped.
+func QuietWrap(err error, format string, args ...any) error {
 	if err == nil {
 		return nil
 	}
 
+	return doWrap(err, true, format, args...)
+}
+
+func doWrap(err error, quiet bool, format string, args ...any) error {
+	var note string
+
 	if len(args) > 0 {
 		note = fmt.Errorf(format, args...).Error()
```

Comment:
Here I agree with the Bot, `fmt.Errorf(...).Error()` is more convoluted than `fmt.Sprintf(...)`. Possibly there is also a small speed gain by doing less conversions but that is not the issue. IMO the way you did it hides the fact that note is a simple string and not an `error` so when I read the code I might get confused.

Reply:
That is for %w support, fmt.Sprintf() doesn't handle it.

Reply:
%w support dropped, it doesn't work as I thought anyway,